lib/spack/spack/test/cmd/compiler_command.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import pytest
+
+import os
+import os.path
+
+import spack.main
+import spack.compilers
+
+compiler = spack.main.SpackCommand('compiler')
+
+
+@pytest.fixture
+def no_compilers_yaml(mutable_config, monkeypatch):
+    """Creates a temporary configuration without compilers.yaml"""
+
+    for scope, local_config in mutable_config.scopes.items():
+        compilers_yaml = os.path.join(
+            local_config.path, scope, 'compilers.yaml'
+        )
+        if os.path.exists(compilers_yaml):
+            os.remove(compilers_yaml)
+
+    # This is essential, otherwise the cache will create weird side effects
+    # that will compromise subsequent tests
+    monkeypatch.setattr(spack.compilers, '_cache_config_file', [])
+
+
+@pytest.mark.regression('11678')
+@pytest.mark.requires_executables('/usr/bin/gcc')
+def test_compiler_find_without_paths(no_compilers_yaml):
```

Comment:
Could you make a temporary directory with the required executable (or rather some executable that behaves as needed)? I thought this would be straightforward because I was assuming some logic like this would already exist in the compiler tests, but admittedly I don't see it there. If that seems like a lot of work for no gain, let me know.

Reply:
> If that seems like a lot of work for no gain, let me know.

I am quite undecided. I see that having a mock executable would be more portable. On the other hand it requires us to reproduce some behavior of compilers that can change over time. 

Given that requiring to have `/usr/bin/gcc` in the test environment doesn't add any additional request to what we already have on Travis, can we postpone this?

Reply:
Yes

Reply:
@alalazo @scheibelp this causes the tests to fail when run on a Mac that hasn't been modified to add `gcc`. I'm looking into a fix.

Reply:
Just to give a little more context, this is because apple computers by default have an executable named gcc which is actually secretly clang. No clue who thought that was a good idea.